server/py/services/api/migrations/versions/c25e56faecce_add_partition_key_column_for_alerts_.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import sqlalchemy as sa
+from alembic import op
+
+import mlrun.common.schemas.partition_interval
+
+import framework.db.sqldb.sql_types
+
+revision = "c25e56faecce"
+down_revision = "6e8e4df16a4e"
+
+TABLE_NAME = "alert_activations"
+_PK_RAW = "pk_alert_activations"
+
+
+def _fill_partition_keys(
+    connection, interval: mlrun.common.schemas.partition_interval.PartitionInterval
+) -> None:
+    """Populate partition_key for all existing rows."""
+    metadata = sa.MetaData()
+    alert_table = sa.Table(
+        TABLE_NAME,
+        metadata,
+        autoload_with=connection,
+    )
+
+    select_stmt = sa.select(alert_table.c.id, alert_table.c.activation_time)
+    for row_id, act_time in connection.execute(select_stmt):
+        connection.execute(
+            alert_table.update()
+            .where(
+                (alert_table.c.id == row_id)
+                & (alert_table.c.activation_time == act_time)
```

Comment:
Why do you need the 2nd filter? The row ID should be enough to uniquely identify the row.